### ISSUE TITLE
Allow customizing the `currentServer` config value at runtime

### ISSUE THREAD

**Issue by networkException** [elk-zone/elk#1427]
Hello! As far as I understand `currentServer` can only be changed at compile time currently which makes self hosting using the container image a lot less practical.

It would be great if a simple

```yml
environment:
    CURRENT_SERVER: host
```

in the compose file would update what the web interface uses

**Comment by danielroe** [comment]
You can set `NUXT_PUBLIC_DEFAULT_SERVER` variable to change the default server.

**Comment by networkException** [comment]
I had no luck with setting that in the compose file's environment:

```yml
environment:
    NUXT_PUBLIC_DEFAULT_SERVER: host
```

Neither by hacking the start command

```yml
command: ["sh", "-c", "NUXT_PUBLIC_DEFAULT_SERVER=host node .output/server/index.mjs"]
```

or by mounting an env file containing

```ini
NUXT_PUBLIC_DEFAULT_SERVER=host
```

to `/elk/.env` using

```yml
volumes:
    - ./.env:/elk/.env:ro
```

**Comment by danielroe** [comment]
Nuxt won't read any .env file. You may need to expose the variable in the Dockerfile.

**Comment by networkException** [comment]
Like so? https://github.com/networkException/elk/commit/142ad835745a68382c9e9887e548321c13270191

Building and running that doesn't work for me, it still defaults to `m.webtoo.ls`

**Comment by networkException** [comment]
Moving that up before any build steps get run *does* work (see https://github.com/networkException/elk/commit/11e632662e59019dde849a5410a1778adb220436), which lead me to conclude that the variable gets set at build time (hence the feature request)

(probably?) see also https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

> This loads process.env.NEXT_PUBLIC_ANALYTICS_ID into the Node.js environment automatically, allowing you to use it anywhere in your code. The value will be inlined into JavaScript sent to the browser because of the NEXT_PUBLIC prefix. This inlining occurs at build time, so your various NEXT_PUBLIC envs need to be set when the project is built.

**Comment by danielroe** [comment]
Nuxt/Nitro _should_ respect this value when set at runtime. The fact that it isn't looks like a bug to me.

**Comment by jibsaramnim** [comment]
It's been a while since I last used Nuxt so I could be thinking about it the wrong way, but [the example shown in their documentation](https://nuxtjs.org/docs/directory-structure/nuxt-config#publicruntimeconfig) shows it might be required to specifically load in an environment variable and fall back to whatever default you might want to use.

Is it possible the automatic passing through of `NUXT_PUBLIC_*` environment variables has gone away since Nuxt moved away from dotenv to using the newer $config component?

**Comment by gmemstr** [comment]
Just want to echo this - I think the priority should be bumped given many instances are getting this up and running for their own users, and this environment variable not working correctly is tripping a lot of people up. Is this something that needs to be set at build time, or run time? Evidence suggests neither is working, and from my own investigations this holds true.

**Comment by danielroe** [comment]
I think I see the issue. In order for this to be configurable, you need to have the value set before the rendering of any page. So if you use the default config in this current repo, `/` is prerendered, meaning any change you make to the default server later on won't take effect on that page (which is probably the one that really counts).

That's why the distributed Dockerfile isn't working for people.

You should be able to work around this in local builds with `NUXT_PUBLIC_DEFAULT_SERVER=bob.com pnpm build`. It might be worth instead avoiding prerendering and instead using rigorous ssr caching.

**Comment by gmemstr** [comment]
For now I've forked the repo and built a custom Docker image with the variable set - not ideal but it works :) (I've also made a few tweaks to the image to work better with our own deployment environment, so win-win I suppose!)

**Comment by loxK** [comment]
> You should be able to work around this in local builds with `NUXT_PUBLIC_DEFAULT_SERVER=bob.com pnpm build`. It might be worth instead avoiding prerendering and instead using rigorous ssr caching.

It doesn't work for me. I altered the Dockerfile like so :

```
# Build
RUN NUXT_PUBLIC_DEFAULT_SERVER=toot.gaiac.io pnpm build
```

And rebuilt :

```
docker-compose up --force-recreate --build -d --remove-orphans
docker image prune -f
```

The default server for local and federated feeds is still the default.

**Comment by jackyzy823** [comment]
Seems related to:

https://github.com/nuxt/nuxt/issues/23695
https://github.com/unjs/nitro/issues/1830

**Comment by maybeanerd** [comment]
It's definitely related to those above mentioned issues!
The question is, will it be solved once the nitro fix is merged into nuxt and then into elk? It might still be "stuck" with the value the prerendering had, as Daniel noted.

We should definitely verify/check that once the fixes have arrived

**Comment by jackyzy823** [comment]
Although nitro and nuxt are updated with those fixes and elk includes new nuxt version, it seems the problem isn't fixed yet.

The problem is that the page is prerendered , so the env `NUXT_PUBLIC_DEFAULT_SERVER` doesn't work at run-time.


You could either 

1) set prerender to `false` here https://github.com/elk-zone/elk/blob/89e3582dd75c232a45d254ba3358876298598217/nuxt.config.ts#L174 , then the env will work at run-time  

2) define `ENV NUXT_PUBLIC_DEFAULT_SERVER=` to prerender the page with `NUXT_PUBLIC_DEFAULT_SERVER` in builder step of docker file  
Note: runtime env still doesn't work 
 https://github.com/elk-zone/elk/blob/89e3582dd75c232a45d254ba3358876298598217/Dockerfile#L6

**Comment by maybeanerd** [comment]
Another option is to "fork it, if you want to keep the prerendering.

As we wanted to adjust the theming as well, that's what we ended up doing:
https://github.com/maybeanerd/crab

Of course it would still be great to have a working envvar to use, since not everyone needs to customize that much.
